### ISSUE TITLE
Build broken

### ISSUE THREAD

**Issue by SoftCreatR** [nicklima/strapi-blog-frontend-next#9]
As mentioned in #7 the current build is broken:

```
Failed to compile.

./pages/_app.tsx:16:8
Type error: 'GlobalStyle' cannot be used as a JSX component.
  Its instance type 'Component<ThemedGlobalStyledClassProps<{}, DefaultTheme>, any, any>' is not a valid JSX element.
    The types returned by 'render()' are incompatible between these types.
      Type 'import("/node_modules/@types/styled-components/node_modules/@types/react/index").ReactNode' is not assignable to type 'React.ReactNode'.
        Type '{}' is not assignable to type 'ReactNode'.

  14 |   return (
  15 |     <GlobalContext.Provider value={global.attributes}>
> 16 |       <GlobalStyle />
     |        ^
  17 |       <Component {...pageProps} />
  18 |     </GlobalContext.Provider>
  19 |   )
error Command failed with exit code 1.
```

**Comment by nicklima** [comment]
I have noticed that. I'm searching for how to fix it. I was about to create an Issue about that but you were faster than me. LOL

**Comment by nicklima** [comment]
@SoftCreatR There's also a problem in `_document.tsx`. I'm trying to fix (without success) both errors.
If you have some directions, I will be happy to hear.

```
Failed to compile.

./pages/_document.tsx:11:7
Type error: Class static side 'typeof MyDocument' incorrectly extends base class static side 'typeof Document'.
  The types returned by 'getInitialProps(...)' are incompatible between these types.
    Type 'Promise<{ styles: Element; html: string; head?: (Element | null)[] | undefined; }>' is not assignable to type 'Promise<DocumentInitialProps>'.
      Type '{ styles: JSX.Element; html: string; head?: (JSX.Element | 
null)[] | undefined; }' is not assignable to type 'DocumentInitialProps'.
        Type '{ styles: JSX.Element; html: string; head?: (JSX.Element 
| null)[] | undefined; }' is not assignable to type '{ styles?: ReactFragment | ReactElement<any, string | JSXElementConstructor<any>>[] | undefined; }'.
          Types of property 'styles' are incompatible.
            Type 'Element' is not assignable to type 'ReactFragment | ReactElement<any, string | JSXElementConstructor<any>>[] | undefined'.  

   9 | } from "next/document"
  10 | 
> 11 | class MyDocument extends Document {
     |       ^
  12 |   static async getInitialProps(ctx: DocumentContext) {
  13 |     const sheet = new ServerStyleSheet()
  14 |     const originalRenderPage = ctx.renderPage
```

**Comment by SoftCreatR** [comment]
> @SoftCreatR There's also a problem in `_document.tsx`. I'm trying to fix (without success) both errors. If you have some directions, I will be happy to hear.
> 
> ```
> Failed to compile.
> 
> ./pages/_document.tsx:11:7
> Type error: Class static side 'typeof MyDocument' incorrectly extends base class static side 'typeof Document'.
>   The types returned by 'getInitialProps(...)' are incompatible between these types.
>     Type 'Promise<{ styles: Element; html: string; head?: (Element | null)[] | undefined; }>' is not assignable to type 'Promise<DocumentInitialProps>'.
>       Type '{ styles: JSX.Element; html: string; head?: (JSX.Element | 
> null)[] | undefined; }' is not assignable to type 'DocumentInitialProps'.
>         Type '{ styles: JSX.Element; html: string; head?: (JSX.Element 
> | null)[] | undefined; }' is not assignable to type '{ styles?: ReactFragment | ReactElement<any, string | JSXElementConstructor<any>>[] | undefined; }'.
>           Types of property 'styles' are incompatible.
>             Type 'Element' is not assignable to type 'ReactFragment | ReactElement<any, string | JSXElementConstructor<any>>[] | undefined'.  
> 
>    9 | } from "next/document"
>   10 | 
> > 11 | class MyDocument extends Document {
>      |       ^
>   12 |   static async getInitialProps(ctx: DocumentContext) {
>   13 |     const sheet = new ServerStyleSheet()
>   14 |     const originalRenderPage = ctx.renderPage
> ```

Should be fixed by #10
